gaboon/commands/install.py

```diff
@@ -1,49 +1,301 @@
 from argparse import Namespace
+from pathlib import Path
+import shutil
 import subprocess
 from packaging.requirements import Requirement
 from gaboon.config import get_config
-from gaboon._dependency_helpers import get_base_install_path
 from gaboon.logging import logger
+import os
+from base64 import b64encode
+import re
+import requests
+import sys
+import traceback
+from tqdm import tqdm
+import zipfile
+from io import BytesIO
+from gaboon.constants.vars import (
+    REQUEST_HEADERS,
+    PACKAGE_VERSION_FILE,
+)
+import tomllib
+import tomli_w
+from enum import Enum
+from urllib.parse import quote
+
+
+class DependencyType(Enum):
+    GITHUB = "github"
+    PIP = "pip"
 
 
 def main(args: Namespace):
     requirements = args.requirements
+    config = get_config()
     if len(requirements) == 0:
-        requirements = get_config().get_dependencies()
-    _pip_install(requirements, args.quiet)
-
-
-def _pip_install(package_ids: list[str], quiet: bool = False):
-    path = get_base_install_path()
-    for i in range(len(package_ids)):
-        if "/" in package_ids[i]:
-            package_ids[i] = (
-                package_ids[i].split("/")[-1]
-                + " @ git+https://github.com/"
-                + package_ids[i]
+        requirements = config.get_dependencies()
+    if len(requirements) == 0:
+        logger.info("No dependencies to install.")
+        return 0
+
+    pip_requirements = []
+    github_requirements = []
+    for requirement in requirements:
+        if classify_dependency(requirement) == DependencyType.GITHUB:
+            github_requirements.append(requirement)
+        else:
+            pip_requirements.append(requirement)
+    install_path: Path = config.get_base_dependencies_install_path()
+    if len(pip_requirements) > 0:
+        _pip_installs(pip_requirements, install_path, config.installer, args.quiet)
+    if len(github_requirements) > 0:
+        _github_installs(github_requirements, install_path, args.quiet)
+    return 0
+
+
+def classify_dependency(dependency: str) -> DependencyType:
+    # GitHub dependency pattern
+    github_pattern = r"^([a-zA-Z0-9_-]+)/([a-zA-Z0-9_-]+)(@[a-zA-Z0-9_.-]+)?$"
```

Comment:
maybe actually just search for the `@` character, i don't think it is allowed in pep440 style requirements

Reply:
https://peps.python.org/pep-0440/#direct-references

Your suggestion was what I originally had in mind; then, while playing with weird stuff, I actually got `uv` to put an `@` sign in the dependencies list. Let me know if I'm confused.

Reply:
hmm yea it can appear in git links as well